### DISCUSSION TITLE
How to get a cleaner homepage?

### DISCUSSION THREAD

**Discussion by Gitonthegrid** [koreader/koreader#10965] [General]
Hi all, 

I am totally new with Koreader, and love it already on my Kobo Libra H20

The only 'problem'  I have, is with quick acces to all my books with covers. 
On the normal kobo software, you could always tap in the middle of the screen, and directly go to "My Books"
It then would take you to all your books and see the covers. 

I Koreader I cant find any option in the gesture menu to have the same type of option. 
Besides that, if I do go to the home menu, I see all folder structures with the author names and within that folders are the books. I rather want to see all my books with the covers directly like it used to be with the kobo reader software. 

Is it possible to get the same sort of layout? Also I would like to disable to go to the parent folder of the Ereader. Ik never have to go there, and if I want to go there, I will plug in into the PC. 

I hope someone can help me, because the rest of the setting are pretty clear to me

**Comment by Frenzie** [comment]
The program's built around folders I'm afraid, though I think @poire-z had a proposal somewhere.

**Comment by hius07** [comment]
> Also I would like to disable to go to the parent folder of the Ereader

Set a folder with your books as Home folder.
Then, while in the file browser, go to the upper menu - the leftmost tab - Settings - Home folder settings - Lock home folder.

**Comment by Galunid** [comment]
> Besides that, if I do go to the home menu, I see all folder structures with the author names and within that folders are the books. I rather want to see all my books with the covers directly like it used to be with the kobo reader software.

This is how Calibre stores the books. You can change it in `Preferences -> Sending books to devices`

![image](https://github.com/koreader/koreader/assets/10298730/3163ad11-78ae-44d1-b19e-77864cc980ec)

In the field highlighted in red put `{title}`. Just beware this will work for future uploads, and won't touch books you already put on your device.

**Reply by Galunid** [reply]
Putting all books in one folder should give you what you want, you can enable cover mode in File manager somewhere ;)

**Reply by Frenzie** [reply]
Note that it's not generally recommended to put more than a few hundred books in one folder for reasons of both performance and navigation.

**Comment by Gitonthegrid** [comment]
@Galunid thank you for the explanation.

So if I just change that setting to "title" in calibre, then they store all epubs in one folder?

And I can just delete the old books en re-import them, that is no problem.

@hius07 thank you, I will take a look at the setting when I am back home. It then works like I want with the setting above from @Galunid ?

**Reply by Galunid** [reply]
> So if I just change that setting to "title" in calibre, then they store all epubs in one folder?

Yes, it should, though you need to remember it's `{title}` - not `title`, the brackets are really important here.

Then in file manager you can set `First tab -> Display mode -> Mosaiac with cover images`. It will give you what you want.

**Reply by Gitonthegrid** [reply]
thank you, that worked the way I wanted, but I can already see why they use the folder structure. It is a bit slow with loading the covers. 

I did also set up my other Ereader with the {authors}/{title} but then I have a other problem. 
Some authors write there books with others like John Grisham. So although John Grisham is the main writer of the book, Calibre will put it not in the "John Grisham" Folder, but instead great a other folder with the second writer. Is there some option in Calibre that will put everything in the same author folder?

**Reply by Galunid** [reply]
No idea, I think a better place to ask is Calibre's forum on [https://www.mobileread.com/forums/forumdisplay.php?f=166](https://www.mobileread.com/forums/forumdisplay.php?f=166).

If you are just interested in organizing your library, and you're alright with using tags, you can try this small script I wrote
https://gist.github.com/Galunid/fe82c95b035e94a6373b0b0309b7adb0. It works based on tags. 

```python
# This is the only part you edit
hierarchy = {
    "Magazine": {
        "%series%": []
    },
    "Chess": {
        "%series%": []
    },
    "IT": {
        "Programming": {
            "Application Design": [],
            "Python": [],
            "JavaScript": [],
            "Java": [],
            "C": [],
            "C++": [],
            "Algorithms": []
        },
        "Operating Systems": [],
        "Algorithms": [],
        "Networking": [],
        "Project Management": [],
        "Data Science": [],
        "Machine Learning": [],
        "InfoSec": [],
        "Databases": [],
        "VCS": []
    },
    "LightNovel": {
        "%series%": [],
    },
    "WebNovel": {
        "%series%": [],
    },
    "Fiction": {
        "%series%": [],
    },
    "Science": [],
    "Psychology": [],
    "Philosophy": [],
    "Business": [],
    "Life Style": [],
    "Mathematics": [],
    "Classics": [],
    "Research Paper": {
        "%author%": []
    }
}
```

It creates folders based on tags. So if a book has the `Programming, IT, Java` tags, it's going to end up in a `IT/Programming/Java/` folder. If no rule matches, it goes to `Other` folder. So if a book has `Programming, IT, Project Management` tags, it'll end up in `IT/Programming/Other/` directory. If you want to access calibre field like `series`, or `author`, surround it with `%` sign -> `%series%`, `%author%`, it should work similarly for other fields. The "higher" the tag is, the more priority it has. So if something is tagged as `IT, Magazine` it'll end up in `Magazine` folder because it's "higher".

Run this script with `python <path to script>`, copy the output and in Calibre go to `Preferences -> Add your own columns -> Add custom column -> Lookup names: output_path -> Column heading: OutputPath -> Column type: Column build from other columns -> Template: what you copied from the program`

In `Send to devices` you set `{#output_path}` instead of `{title}` or author or whatever is there right now.

That's enough off topic for now ;)
